lambda/orc_ingestion.py

```diff
@@ -58,27 +60,43 @@ def lambda_handler(event, context):
 
 
 def get_nhs_number_list_from_s3(filename: str, bucket_name: str) -> list:
-    local_file_path = f"/tmp/{filename}"
-    s3_client.download_file(
-        bucket_name, filename, local_file_path
+    response = s3_client.get_object(
+        Bucket=bucket_name,
+        Key=filename
     )
-    with open(local_file_path, "r") as f:
-        nhs_number_list = f.read()
-    os.remove(local_file_path)
-    return nhs_number_list.split(',')
+
+    file_contents = response["Body"].readlines()
+
+    if len(file_contents) > 1:
+        raise InvalidFileFormatException(
+            "All NHS numbers must be contained on a single line, seperated by commas."
+        )
+
+    nhs_numbers_str = file_contents[0].decode('utf-8')
+
+    only_nums_and_commas_bool = bool(re.compile(r'[0-9,]').match(nhs_numbers_str))
```

Comment:
could we test to see if this line works when the re.complie is removed as I think the compile may be done by the match.

suggested code:
only_nums_and_commas_bool = bool(re.match('^[0-9,]+$', nhs_numbers_str))

Reply:
regex from this python example

![image](https://github.com/nhsconnect/prm-repo-suspension-service/assets/127425466/afd102fa-db77-41f5-89dd-575845acd7e1)